xproj/index.py

```diff
@@ -58,13 +58,22 @@ def from_variables(
         if len(variables) != 1:
             raise ValueError("can only create a CRSIndex from one scalar variable")
 
+        varname = next(iter(variables.keys()))
         var = next(iter(variables.values()))
 
         if var.ndim != 0:
             raise ValueError("can only create a CRSIndex from one scalar variable")
 
-        # TODO: how to deal with different CRS in var attribute vs. build option?
-        crs = var.attrs.get("spatial_ref", options["crs"])
+        try:
+            if "crs" in options:
+                crs = pyproj.CRS.from_user_input(options["crs"])
+            else:
+                crs = pyproj.CRS.from_cf(var.attrs)
```

Comment:
```suggestion
            elif "spatial_ref" in var.attrs:
                pyproj.CRS.from_user_input(var.attrs["spatial_ref"])
            else:
                crs = pyproj.CRS.from_cf(var.attrs)
```

?

Reply:
IIUC It is for compatibility with GDAL (same expected value as `crs_wkt`).

Reply:
Turns out `CRS.from_cf` handles that, I added a test.